df.history.xml

```diff
@@ -459,25 +460,25 @@
                         <enum type-name='reputation_type' comment="Probably ordered"/>
                     </stl-vector>
                     <stl-vector name="counter" type-name='int32_t' comment="One element for each 'attitude' element. Guess 0 - 100"/>
-                    <int16_t name="rank" since='v0.47.01'/>
-                    <int32_t name='loyalty' comment="0 - 100"/>
+                    <int32_t name='rank' init-value='65535' comment="seems to be fixed. Might not be used anymore" since='v0.47.01'/>
+                    <int32_t name='loyalty' comment="0 - 100. Loyalty, Respect, Fear, and Trust values do not seem to affect the relationship screen description, only Love and Familiarity"/>
                     <int32_t name='respect' comment="-100 - 100"/>
                     all 0xfff0bdc0 = -1000000
                     <int32_t name='fear' comment="-100 - 100"/>
-                    <int32_t name='love' comment="-100 - 100"/>
+                    <int32_t name='love' comment="-100 - 100. Affects relation description: -100: Pure Hate, LE -75: Hated, LE -50: Disliked, LE 49: Acquaintance, LE 74: Friend, LE 99: Close Friend, 100: Kindred Spirit. Lower histfig_id is sorted above higher, except Acquaintance which is sorted on Familiarity"/>
                     <int32_t name='trust' comment="-100 - 100"/>
-                    <int32_t comment="0 - 60 seen"/>
+                    <int32_t name='familiarity' comment="Shifts from 'Passing' to 'Long-term' Acquaintance at 30, but a Love value other than Acquaintance takes presedence. Higher Familiarity is sorted above lower, with lower histfig_id is sorted above higher on a tie"/>
                     <int32_t name='last_meet_year' comment="probably latest contact. Seen -1"/>
                     <int32_t name='last_meet_seconds72'/>
                     <int32_t name='meet_count' since='v0.44.10'/>
-                    <int32_t since='v0.44.10'/>
-                    <int32_t since='v0.44.10'/>
-                    <int32_t since='v0.47.01'/>
-                    <int32_t since='v0.47.01'/>
-                    <int32_t since='v0.47.01'/>
-                    <int32_t since='v0.47.01'/>
-                    <int32_t comment='only seen 0' since='v0.47.01'/>
-                    <int32_t comment='only seen 0' since='v0.47.01'/>
+                    <int32_t name='unk_2' since='v0.44.10'/>
+                    <int32_t name='unk_3' since='v0.44.10'/>
+                    <int32_t name='unk_4' comment='probably uninitialized' since='v0.47.01'/>
+                    <int32_t name='unk_5' comment='probably uninitialized' since='v0.47.01'/>
+                    <int32_t name='unk_6' comment='probably uninitialized' since='v0.47.01'/>
+                    <int32_t name='unk_7' comment='probably uninitialized' since='v0.47.01'/>
+                    <int32_t name='unk_8' comment='only seen 0' since='v0.47.01'/>
+                    <int32_t name='unk_9' comment='only seen 0' since='v0.47.01'/>
```

Comment:
The 47 branch was used for a steady stream of small unrelated changes, intended to be possible to merge at any time (the only "outstanding" cases are the ones where a conflict is detected when the updates are pushed, and they're handled as quickly as possible after detection. What I'd hoped for, and what happened early on, was that it would be merged with reasonably short intervals so there wouldn't be such a mountain building up (getting increasingly burdensome as parallel changes clash with it). I'm certainly aware of the "this is too much effort to tackle now, I need a bigger chunk of time to do that" issue.
It's certainly possible that you'd make a merge and a new change trickles in minutes later, but the new change should be (mostly) unrelated to the previous ones, so it's not a big deal if it ends up waiting until the next merge opportunity.

So far I've kept my drafts local, so I haven't had any need to flag them (I've seen others using that, though, and it's useful when you want/need comments).

It's good that the fields listed have been identified as a vector, which means my comments are no longer relevant. The comments state what I saw (as pointers look like changing data), but I don't remember things in such details as to be able to comment on the comments beyond that.

Reply:
Got it. I'll do my best to keep up with them faster.
I forgot to go into more detail earlier: `devel/visualize-structures` (and the `memview` plugin, if you build with dev plugins enabled) marks valid pointers, so that could be helpful to tell whether random-looking data is actually a pointer (since vectors consist of three consecutive pointers). The `vectors` plugin will do some additional checks to find valid vectors. None of these will deal with empty vectors with 3 null pointers, though.

Reply:
I actually just ran into some of these fields populated in a pocket-sized world. Turns out they're not vectors after all, so you were right. (They're not even valid pointers.) I'll spin up a 32-bit instance to investigate whether they're intptr_ts or int32_ts.

There's a more complete list of memory-research tools here, in case it's useful: https://docs.dfhack.org/en/latest/docs/Memory-research.html

Reply:
Thanks for digging deeper into this.